extensions/replication/tasks/UpdateReplicationStatus.js

```diff
@@ -30,166 +24,33 @@ class UpdateReplicationStatus extends BackbeatTask {
-            endpoint: `${this.sourceConfig.transport}://` +
-                `${sourceS3.host}:${sourceS3.port}`,
-            credentials: this.s3sourceCredentials,
-            sslEnabled: this.sourceConfig.transport === 'https',
-            httpOptions: { agent: this.sourceHTTPAgent, timeout: 0 },
-            maxRetries: 0,
-        });
+        this.backbeatSourceClient =
+            new BackbeatMetadataProxy(this.sourceConfig, this.sourceHTTPAgent);
     }
 
     processQueueEntry(sourceEntry, done) {
         const log = this.logger.newRequestLogger();
 
         log.debug('updating replication status for entry',
                   { entry: sourceEntry.getLogInfo() });
-
-        const err = this._setupSourceRole(sourceEntry, log);
-        if (err) {
-            return setImmediate(() => done(err));
+        const { error } =
+            this.backbeatSourceClient.setupSourceRole(sourceEntry, log);
+        if (error) {
+            return setImmediate(() => done(error));
         }
+        this.backbeatSourceClient.setSourceClient(log);
         return this._updateReplicationStatus(sourceEntry, log, done);
     }
 
-    _getMetadata(entry, log, cb) {
-        this.retry({
-            actionDesc: 'get metadata from source',
-            logFields: { entry: entry.getLogInfo() },
-            actionFunc: done => this._getMetadataOnce(entry, log, done),
-            shouldRetryFunc: err => err.retryable,
-            log,
-        }, cb);
-    }
-
-    _getMetadataOnce(entry, log, cb) {
-        log.debug('getting metadata', {
-            where: 'source',
-            entry: entry.getLogInfo(),
-            method: 'ReplicateObject._getMetadataOnce',
-        });
-        const cbOnce = jsutil.once(cb);
-
-        const req = this.backbeatSource.getMetadata({
-            Bucket: entry.getBucket(),
-            Key: entry.getObjectKey(),
-            VersionId: entry.getEncodedVersionId(),
-        });
-        attachReqUids(req, log);
-        req.send((err, data) => {
-            if (err) {
-                // eslint-disable-next-line no-param-reassign
-                err.origin = 'source';
-                if (err.ObjNotFound || err.code === 'ObjNotFound') {
-                    return cbOnce(err);
-                }
-                log.error('an error occurred when getting metadata from S3', {
-                    method: 'ReplicateObject._getMetadataOnce',
-                    entry: entry.getLogInfo(),
-                    origin: 'source',
-                    error: err,
-                    errMsg: err.message,
-                    errCode: err.code,
-                    errStack: err.stack,
-                });
-                return cbOnce(err);
-            }
-            return cbOnce(null, data);
-        });
-    }
-
     _refreshSourceEntry(sourceEntry, log, cb) {
-        this._getMetadata(sourceEntry, log, (err, blob) => {
+        const params = {
+            logInfo: sourceEntry.getLogInfo(),
```

Comment:
I think it's better not to pass `logInfo` and let the function log whatever it feels good to log. (same for other similar cases)

Reply:
I'm not sure about this since that method `getLogInfo` was determining what was being logged before. Do you mean just keep calling `getLogInfo` in the helper method itself rather than passing it as a param?

Reply:
No, `getLogInfo` is used as a helper whenever an `entry` is passed to log its relevant info, but since we don't pass `entry` anymore there's no point in passing this as a `logInfo` param. I would suggest to just log relevant info directly (say, `bucket` and `objectKey`).

Reply:
Okay, I see what you mean. I'll update this.